content/en/docs/guides/performance.md

```diff
@@ -0,0 +1,102 @@
+---
+title: Performance Best Practices
+description: A user guide of both general and language-specific best practices to improve performance.
+---
+
+### General
+
+*   Always **re-use stubs and channels** when possible.
+
+*   **Use keepalive pings** to keep HTTP/2 connections alive during periods of
+    inactivity to allow initial RPCs to be made quickly without a delay (i.e.
+    C++ channel arg GRPC_ARG_KEEPALIVE_TIME_MS). 
+
+*   **Use streaming RPCs** when wanting to avoid continuous RPC initiation,
+    which includes connection load balancing at the client-side, starting a new
+    HTTP/2 request at the transport layer, and invoking a user-defined method
+    handler on the server side. Streams are particularly well-suited to handling
+    a long-lived logical flow of data from the client-to-server,
+    server-to-client, or in both directions. Streams, however, cannot be load
+    balanced once they have started and can be hard to debug for stream
+    failures. \
+    ***Side note:*** *This does not apply to Python (see Python section for
+    details).*
+
+*   Each gRPC channel uses 0 or more HTTP/2 connections and each connection
+    usually has a limit on the number of concurrent streams. When the number of
+    active RPCs on the connection reaches this limit, additional RPCs are queued
+    in the client and must wait for active RPCs to finish before they are sent.
+    Applications with high load or long-lived streaming RPCs might see
+    performance issues because of this queueing. There are two possible
+    solutions:
+
+    1.  **Create a separate channel for each area of high load** in the
+        application, or
+
+    2.  **Use a pool of gRPC channels** to randomly distribute RPCs over
+        multiple connections (channels must have different channel args to
+        prevent re-use so define a use-specific channel arg such as channel
+        number). \
+        ***Side note:*** *The gRPC team has plans to add a feature to fix these
+        performance issues, so any solution involving creating multiple channels
+        is a temporary workaround that should eventually not be needed.*
+
+### C++
+
+*   **Do not use Sync API for performance sensitive servers.** If performance
+    and/or resource consumption are not concerns, use the Sync API as it is the
+    simplest to implement for low-QPS services.
+
+*   **Favor callback API over other APIs for all RPCs**, given that the
```

Comment:
@vjpai Would this apply to the OSS callback API as well?

Reply:
Yes. It's slower than sync at OSS for the time being but we  (well, you) will improve performance and it's only slower for truly high-QPS workloads at the current moment.

Reply:
... and it's actually a usable API unlike CQ-based async

Reply:
@vjpai High-QPS meaning what (a rough estimate is fine)? It seems for the OSS version, we don't want to publicize that the callback API should be favored for all RPCs at this point